src/functionalTest/java/uk/gov/hmcts/reform/jobscheduler/controllers/JobsControllerTest.java

```diff
@@ -26,31 +38,141 @@
     @Test
     public void should_return_empty_list_of_jobs() throws Exception {
-        send("")
+        sendGetRequest()
             .andExpect(status().isOk())
-            .andExpect(content().json(
-                "{"
-                    + "\"content\":[],"
-                    + "\"total_pages\":0,"
-                    + "\"total_elements\":0,"
-                    + "\"last\":true,"
-                    + "\"size\":10,"
-                    + "\"number\":0,"
-                    + "\"sort\":null,"
-                    + "\"first\":true,"
-                    + "\"number_of_elements\":0"
-                    + "}"
-            ))
+            .andExpect(jsonPath("content").isEmpty())
+            .andExpect(jsonPath("total_pages").value(0))
+            .andExpect(jsonPath("total_elements").value(0))
+            .andExpect(jsonPath("last").value(true))
+            .andExpect(jsonPath("size").value(10))
+            .andExpect(jsonPath("number").value(0))
+            .andExpect(jsonPath("sort").value(isNull()))
+            .andExpect(jsonPath("first").value(true))
+            .andExpect(jsonPath("number_of_elements").value(0));
+    }
+
+    @Test
+    public void should_create_a_new_job_and_return_job_uri_of_newly_created_job_when_valid_job_input_is_passed()
+        throws Exception {
+        //Given
+        String jobJson = jobJson();
+
+        //when
+        MvcResult mvcResult = sendPostRequest(jobJson)
+            .andExpect(status().isCreated())
+            .andReturn();
+
+        //then
+        URI newJobUri = new URI(mvcResult.getResponse().getRedirectedUrl());
+
+        assertThat(newJobUri).isNotNull();
+
+        //Delete the newly created job else it will used by other test cases
+        sendDeleteRequest(extractJobId(mvcResult.getResponse().getRedirectedUrl()))
+            .andExpect(status().isNoContent());
+    }
+
+    @Test
+    public void should_not_create_a_new_job_and_throw_4xx_client_error_when_empty_job_input_is_passed()
+        throws Exception {
+        sendPostRequest("")
+            .andExpect(status().is4xxClientError())
             .andReturn();
     }
 
-    private ResultActions send(String content) throws Exception {
+    @Test
+    public void should_delete_job_successfully_when_existing_job_id_is_passed() throws Exception {
+        //Given
+        String jobJson = jobJson();
+
+        //when
+        MvcResult mvcResult = sendPostRequest(jobJson)
+            .andExpect(status().isCreated())
+            .andReturn();
+
+        //then
+        String jobId = extractJobId(mvcResult.getResponse().getRedirectedUrl());
+
+        sendDeleteRequest(jobId)
+            .andExpect(status().isNoContent());
+    }
+
+    @Test
+    public void should_not_delete_job_and_throw_job_not_exception_when_non_existing_job_id_is_passed()
+        throws Exception {
+        MvcResult mvcResult = sendDeleteRequest(UUID.randomUUID().toString())
+            .andExpect(status().is4xxClientError())
```

Comment:
Is there a way to verify the exact response status? We want 404, but not 401.

Reply:
yes `andExpect(status().isNotFound())`